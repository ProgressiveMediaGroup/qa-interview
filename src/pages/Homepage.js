// Node_Modules
import React, { Component } from 'react'

// Display Components
import Body from '../components/Body'

class Homepage extends Component {
  render () {
    return (
      <Body heading='Progressive Content' subheading='Front-end Technical Test'>
        <p>This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.</p>
        <p>For more information, check out the README file...</p>
      </Body>
    )
  }
}

export default Homepage
