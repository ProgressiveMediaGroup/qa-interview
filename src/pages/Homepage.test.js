/* global it, expect, describe */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Homepage from './Homepage'

it('renders without crashing', () => {
  shallow(<Homepage />)
})

it('should match snapshot', () => {
  const wrapper = shallow(<Homepage />)
  expect(wrapper).toMatchSnapshot()
})

describe('Testing Strapline', function () {
  it('H1 Title is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('h1').length).toBe(1)
  })

  it('H1 Title is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('h1').text()).toBe('Progressive Content')
  })

  it('H2 Title is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('h2').length).toBe(1)
  })

  it('H2 Title is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('h2').text()).toBe('Front-end Technical Test')
  })
})

describe('Testing Content', function () {
  it('P tag is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('p').length).toBe(2)
  })

  it('First P tag is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('p').at(0).text()).toBe('This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.')
  })

  it('Second P tag is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    )
    expect(wrapper.find('p').at(1).text()).toBe('For more information, check out the README file...')
  })
})
