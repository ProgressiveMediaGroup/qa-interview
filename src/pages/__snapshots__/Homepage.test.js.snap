// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <p>
          This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.
</p>,
        <p>
          For more information, check out the README file...
</p>,
      ],
      "heading": "Progressive Content",
      "subheading": "Front-end Technical Test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.",
        },
        "ref": null,
        "rendered": "This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "For more information, check out the README file...",
        },
        "ref": null,
        "rendered": "For more information, check out the README file...",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <p>
            This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.
</p>,
          <p>
            For more information, check out the README file...
</p>,
        ],
        "heading": "Progressive Content",
        "subheading": "Front-end Technical Test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.",
          },
          "ref": null,
          "rendered": "This project is used as a technical test when interviewing potential front-end web developers for Progressive Content.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "For more information, check out the README file...",
          },
          "ref": null,
          "rendered": "For more information, check out the README file...",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
